datapipelines/src/main/kotlin/io/customer/sdk/DataPipelineInstance.kt

```diff
@@ -85,6 +85,74 @@ abstract class DataPipelineInstance : CustomerIOInstance {
         serializationStrategy: SerializationStrategy<Traits>
     )
 
+    /**
+     * The track method helps you record events: the things your users do on your app.
+     * Each track call records a single event. Each event has a name and properties.
+     * For example, if you send a track call when someone starts a video on your app,
+     * the name of the event might be Video Started,
+     * and the properties might include the title of the video, the length of the video, and so on.
+     *
+     * @param name Name of the action
+     * @param properties custom values providing extra information about the event.
+     * @see [Learn more](https://customer.io/docs/cdp/sources/source-spec/track-spec/)
+     */
+    @JvmOverloads
+    fun track(name: String, properties: JsonObject = emptyJsonObject) {
+        track(name, properties, JsonAnySerializer.serializersModule.serializer())
+    }
+
+    /**
+     * The track method helps you record events: the things your users do on your app.
+     * Each track call records a single event. Each event has a name and properties.
+     * For example, if you send a track call when someone starts a video on your app,
+     * the name of the event might be Video Started,
+     * and the properties might include the title of the video, the length of the video, and so on.
+     *
+     * @param name Name of the action
+     * @param properties Map of <String, Any> to be added
+     * @see [Learn more](https://customer.io/docs/cdp/sources/source-spec/track-spec/)
+     */
+    fun track(name: String, properties: CustomAttributes) {
```

Comment:
Same questions as screen changes. I think we decided to use `title` here instead.

```suggestion
    fun track(title: String, properties: CustomAttributes) {
```